backend_test.go

```diff
@@ -207,33 +214,41 @@ func (m *mockProvider) DeleteApplication(ctx context.Context, applicationObjectI
 	return autorest.Response{}, nil
 }
 
-func (m *mockProvider) UpdateApplicationPasswordCredentials(ctx context.Context, applicationObjectID string, parameters graphrbac.PasswordCredentialsUpdateParameters) (result autorest.Response, err error) {
-	m.passwords = make(map[string]bool)
-	for _, v := range *parameters.Value {
-		m.passwords[*v.KeyID] = true
+func (m *mockProvider) AddApplicationPassword(ctx context.Context, applicationObjectID string, displayName string, endDateTime date.Time) (result PasswordCredentialResult, err error) {
+	keyID := generateUUID()
+	cred := passwordCredential{
+		DisplayName: to.StringPtr(displayName),
+		StartDate:   &date.Time{Time: time.Now()},
+		EndDate:     &endDateTime,
+		KeyID:       to.StringPtr(keyID),
+		SecretText:  to.StringPtr(generateUUID()),
 	}
 
-	return autorest.Response{}, nil
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	m.passwords[keyID] = cred
+
+	return PasswordCredentialResult{
+		passwordCredential: cred,
+	}, nil
 }
 
-func (m *mockProvider) ListApplicationPasswordCredentials(ctx context.Context, applicationObjectID string) (result graphrbac.PasswordCredentialListResult, err error) {
-	var creds []graphrbac.PasswordCredential
-	for keyID := range m.passwords {
-		kCopy := keyID
-		creds = append(creds, graphrbac.PasswordCredential{KeyID: &kCopy})
-	}
+func (m *mockProvider) RemoveApplicationPassword(background context.Context, applicationObjectID string, keyID string) (result autorest.Response, err error) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
 
-	return graphrbac.PasswordCredentialListResult{
-		Value: &creds,
-	}, nil
+	delete(m.passwords, keyID)
+
+	return autorest.Response{}, nil
 }
 
 func (m *mockProvider) appExists(s string) bool {
 	return m.applications[s]
 }
 
 func (m *mockProvider) passwordExists(s string) bool {
-	return m.passwords[s]
+	_, ok := m.passwords[s]
```

Comment:
I think we need to add a `s.lock.RLock()` here though.  https://stackoverflow.com/questions/36167200/how-safe-are-golang-maps-for-concurrent-read-write-operations

Reply:
Good catch.